pkg/flow/flow.go

```diff
@@ -209,9 +214,66 @@ func (c *Flow) LoadFile(f *File) error {
 	return diags.ErrorOrNil()
 }
 
+// ComponentInfos returns the component infos.
+func (c *Flow) ComponentInfos() []*river.ComponentField {
+	cns := c.loader.Components()
+	infos := make([]*river.ComponentField, len(cns))
+	refByBacktrack := make(map[string]*river.ComponentField, 0)
+	for i, com := range cns {
+		refs, _ := controller.ComponentReferences(com, c.loader.Graph())
+		nn := newFromNode(com, refs)
+		infos[i] = nn
+		refByBacktrack[nn.ID] = nn
+	}
+	// We need to backtrack the infos
+	for _, info := range infos {
+		for _, refTo := range info.References {
+			refByBacktrack[refTo].ReferencedBy = append(refByBacktrack[refTo].ReferencedBy, info.ID)
+		}
+	}
+	return infos
+}
+
 // Close closes the controller and all running components.
 func (c *Flow) Close() error {
 	c.cancel()
 	<-c.exited
 	return c.sched.Close()
 }
+
+func newFromNode(cn *controller.ComponentNode, refs []controller.Reference) *river.ComponentField {
+	refsStr := make([]string, len(refs))
+	for i, r := range refs {
+		refsStr[i] = strings.Join(r.Target.ID(), ".")
+	}
+	ci := &river.ComponentField{
+		Field: river.Field{
+			ID:   cn.NodeID(),
+			Name: strings.Join(cn.ID()[0:2], "."),
+			Type: "block",
+			Key:  strings.Join(cn.ID(), "."),
```

Comment:
This worries me a little bit, the first two strings in the ID won't always be the namespace (i.e. `prometheus.integration.node_exporter`.)

I think we should update ComponentNode to store the component name & component label separately instead of us having to extract them from the ID.

Reply:
Good idea.